### PR TITLE
feat(progress): add Progress and ProgressController

### PR DESCRIPTION
**Pull request by joggienl**
The test for the controller are not complete yet. If approved, I'll create an issue to address this.

### CODE REVIEW

**Comment by joggienl** [comment]
I've added some test for the controller. Coverage is back to 100%.

File: src/administrative-sdk/administrative-sdk.js [hunk 1]
```diff
@@ -515,6 +517,20 @@ export default class AdministrativeSDK {
   getGroups() {
     return this._groupController.getGroups();
   }
+
+  /**
+   * Get and return progress on the requested {@Link Category} for the current {@Link User} derived
+   * from the OAuth2 scope.provided categoryId. The progress wil be returned for the current user.
+   * If a user is eligible to see the progress of more user, that that progress is returned as well.
```
Comment (kvdb) [inline src/administrative-sdk/administrative-sdk.js]: users, that progress

File: src/administrative-sdk/progress/progress-controller.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  /**
+   * Get Progress corresponding to a {@Link Category} for the current active {@Link User} derived from
+   * the OAuth2 scope. If a user is allowed to see the progress of multiple users (depending of its
+   * {@Link Role} and permissions) then more Progress objects could be returned.
+   *
+   * Progress is always being fetched for a single category, identified with a categoryId.
+   *
+   * @param {string} categoryId - Specify a Category identifier.
+   * @returns {Promise.<Progress[]>} Array of Progress.
```
Comment (kvdb) [inline src/administrative-sdk/progress/progress-controller.js]: Isn't that something like:
{?Array.<Progress>}

File: src/administrative-sdk/progress/progress.js [hunk 3]
```diff
@@ -1,10 +1,56 @@
...
 export default class Progress {
-  constructor(done, total) {
-    this.done = done;
-    this.total = total;
+  /**
+   * Create a Progress object.
+   *
+   * @param {Object} user - Some user information that belongs to the progress.
+   * @param {string} category - The category identifier to which this progress object refers.
+   * @param {?string} percentage - The percentage of completeness of all (underlying) challenges.
```
Comment (kvdb) [inline src/administrative-sdk/progress/progress.js]: i'd say it's an int or float

File: src/administrative-sdk/progress/progress-controller.js (hunk 2 above)
Comment (joggienl) [inline src/administrative-sdk/progress/progress-controller.js]: Well.. in this case not. A Promise will be returned. And the resolved result will then be an Array with Progress objects. So Promise.<Progress[] is OK here.

As a side note, I looked at other controllers how it was done..

**Pull request merged**